hyperspy/signal.py

```diff
@@ -4542,11 +4554,19 @@ def map(
             res = self._map_all(function, inplace=inplace, **kwargs)
         else:
             # Iteration over coordinates.
-            res = self._map_iterate(function, iterating_kwargs=ndkwargs,
-                                    show_progressbar=show_progressbar,
-                                    parallel=parallel, max_workers=max_workers,
-                                    ragged=ragged, inplace=inplace,
-                                    **kwargs)
+            if self._lazy:
+                res = self._map_iterate(function, iterating_kwargs=ndkwargs,
```

Comment:
Instead of writing twice almost the same, would it be better to have `output_dtype` and `output_signal_size` added to `kwargs` in case of lazy signal and write the `self._map_iterate` only once?